pkg/kubernetes/helm.go

```diff
@@ -0,0 +1,218 @@
+	RunExecutable(e string, p ...string) error
+
+	piperutils.FileUtils
+}
+
+type HelmExecuteOptions struct {
+	AdditionalParameters      []string `json:"additionalParameters,omitempty"`
+	APIServer                 string   `json:"apiServer,omitempty"`
+	ChartPath                 string   `json:"chartPath,omitempty"`
+	ContainerRegistryPassword string   `json:"containerRegistryPassword,omitempty"`
+	ContainerImageName        string   `json:"containerImageName,omitempty"`
+	ContainerImageTag         string   `json:"containerImageTag,omitempty"`
+	ContainerRegistryURL      string   `json:"containerRegistryUrl,omitempty"`
+	ContainerRegistryUser     string   `json:"containerRegistryUser,omitempty"`
+	ContainerRegistrySecret   string   `json:"containerRegistrySecret,omitempty"`
+	DeploymentName            string   `json:"deploymentName,omitempty"`
+	DeployTool                string   `json:"deployTool,omitempty" validate:"possible-values=helm helm3"`
+	ForceUpdates              bool     `json:"forceUpdates,omitempty"`
+	HelmDeployWaitSeconds     int      `json:"helmDeployWaitSeconds,omitempty"`
+	HelmValues                []string `json:"helmValues,omitempty"`
+	Image                     string   `json:"image,omitempty"`
+	IngressHosts              []string `json:"ingressHosts,omitempty"`
+	KeepFailedDeployments     bool     `json:"keepFailedDeployments,omitempty"`
+	KubeConfig                string   `json:"kubeConfig,omitempty"`
+	KubeContext               string   `json:"kubeContext,omitempty"`
+	Namespace                 string   `json:"namespace,omitempty"`
+	TillerNamespace           string   `json:"tillerNamespace,omitempty"`
+	DockerConfigJSON          string   `json:"dockerConfigJSON,omitempty"`
+}
+
+type deployUtilsBundle struct {
+	*command.Command
+	*piperutils.Files
+}
+
+func NewDeployUtilsBundle() HelmDeployUtils {
+	utils := deployUtilsBundle{
+		Command: &command.Command{
+			ErrorCategoryMapping: map[string][]string{
+				log.ErrorConfiguration.String(): {
+					"Error: Get * no such host",
+					"Error: path * not found",
+					"Error: rendered manifests contain a resource that already exists.",
+					"Error: unknown flag",
+					"Error: UPGRADE FAILED: * failed to replace object: * is invalid",
+					"Error: UPGRADE FAILED: * failed to create resource: * is invalid",
+					"Error: UPGRADE FAILED: an error occurred * not found",
+					"Error: UPGRADE FAILED: query: failed to query with labels:",
+					"Invalid value: \"\": field is immutable",
+				},
+				log.ErrorCustom.String(): {
+					"Error: release * failed, * timed out waiting for the condition",
+				},
+			},
+		},
+		Files: &piperutils.Files{},
+	}
+	// reroute stderr output to logging framework, stdout will be used for command interactions
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+// check settings for helm execution
+func runHelmInit(config HelmExecuteOptions, utils HelmDeployUtils, stdout io.Writer) error {
+
+	if len(config.ChartPath) <= 0 {
```

Comment:
could we use conditions in step.yaml instead? -> `mandatoryIf` condition?

Reply:
added it